rust/src/applayernewmodbus/newmodbus.rs

```diff
@@ -25,6 +25,160 @@ use super::parser;
 
 static mut ALPROTO_NEWMODBUS: AppProto = ALPROTO_UNKNOWN;
 
+/* Modbus Protocol version. */
+pub const MODBUS_PROTOCOL_VER: u16 = 0x0000;
+
+/* Modbus Function Code Categories. */
+// TODO - confirm if those should really be u8
+pub const MODBUS_CAT_NONE: u8 =                   0x0;
+pub const MODBUS_CAT_PUBLIC_ASSIGNED: u8 =      1<<0;
+pub const MODBUS_CAT_PUBLIC_UNASSIGNED: u8 =    1<<1;
+pub const MODBUS_CAT_USER_DEFINED: u8 =         1<<2;
+pub const MODBUS_CAT_RESERVED: u8 =             1<<3;
+pub const MODBUS_CAT_ALL: u8 =                   0xFF;
+
+/* Modbus Function Code. */
+pub const MODBUS_FUNC_NONE: u8 =                 0x00;
+
+/* Modbus Read/Write function and Access Types. */
+pub const MODBUS_TYP_NONE: u8 =                  0x0;
+pub const MODBUS_TYP_ACCESS_MASK: u8 =           0x03;
+pub const MODBUS_TYP_READ: u8 =                 1<<0;
+pub const MODBUS_TYP_WRITE: u8 =                1<<1;
+pub const MODBUS_TYP_ACCESS_FUNCTION_MASK: u8 =  0x3C;
+pub const MODBUS_TYP_BIT_ACCESS_MASK: u8 =       0x0C;
+pub const MODBUS_TYP_DISCRETES: u8 =            1<<2;
+pub const MODBUS_TYP_COILS: u8 =                1<<3;
+pub const MODBUS_TYP_WORD_ACCESS_MASK: u8 =      0x30;
+pub const MODBUS_TYP_INPUT: u8 =                1<<4;
+pub const MODBUS_TYP_HOLDING: u8 =              1<<5;
+pub const MODBUS_TYP_SINGLE: u8 =               1<<6;
+pub const MODBUS_TYP_MULTIPLE: u8 =             1<<7;
+pub const MODBUS_TYP_WRITE_SINGLE: u8 =         MODBUS_TYP_WRITE | MODBUS_TYP_SINGLE;
+pub const MODBUS_TYP_WRITE_MULTIPLE: u8 =       MODBUS_TYP_WRITE | MODBUS_TYP_MULTIPLE;
+pub const MODBUS_TYP_READ_WRITE_MULTIPLE: u8 =  MODBUS_TYP_READ | MODBUS_TYP_WRITE | MODBUS_TYP_MULTIPLE;
+
+/* Modbus Read/Write function and Access Types. */
+// TODO I think those exactly the same as the last three above, can we delete them, therefore?
+// pub const MODBUS_TYP_WRITE_SINGLE         MODBUS_TYP_WRITE | MODBUS_TYP_SINGLE;
+// pub const MODBUS_TYP_WRITE_MULTIPLE       MODBUS_TYP_WRITE | MODBUS_TYP_MULTIPLE;
+// pub const MODBUS_TYP_READ_WRITE_MULTIPLE  MODBUS_TYP_READ | MODBUS_TYP_WRITE | MODBUS_TYP_MULTIPLE;
+
+// Modbus Application Protocol Header (MBAP) size
+pub const MODBUS_PDA_HEADER_LEN: usize = 7; 
+
+/* Modbus Protocol Data Unit (PDU) length range. */
+pub const MODBUS_MIN_ADU_LEN: usize =   2;
+pub const MODBUS_MAX_ADU_LEN: usize =  254; 
+// TODO confirm MODBUS PDU MAX value with tests and pcaps, documentation says max is 253
+// Unless it's being used like that because the length field in the MBAP header contains the length
+// including next field, which is unit address (1 byte), in which case, yes, if we compare 
+// with that in mind, max is 254. but, then, min len would be... 3? still confused
+// TODO should this really be usize? I'm having trouble using it, because header length is
```

Comment:
I think usize can be used but maybe its better to use defined types wherever you can? 
Would be interested to know about this more from @jasonish though.

You can use explicit type conversion later in the code to handle operations among variables with different datatypes.

Reply:
I was reading about those types, and also checked what was done in some other of ours (like dcerpc and http2) and saw that in some places usize was chosen, and apparently this is the choice when we are talking about memory size? 
I had some trouble with the type conversion, Rust complained of all my tries, so I'll have to do some research, in case usize is indeed the best choice here.